cmd/installer/pkg/install/manifest.go

```diff
@@ -178,15 +178,16 @@ func (t *Target) Partition(bd *blockdevice.BlockDevice) (err error) {
 
 	switch t.Label {
 	case constants.BootPartitionLabel:
-		// EFI System Partition
-		typeID := "C12A7328-F81F-11D2-BA4B-00A0C93EC93B"
+		// Boot Partition
+		// typeID := "C12A7328-F81F-11D2-BA4B-00A0C93EC93B"
+		typeID := "21686148-6449-6E6F-744E-656564454649"
 		opts = append(opts, partition.WithPartitionType(typeID), partition.WithPartitionName(t.Label), partition.WithLegacyBIOSBootableAttribute(true))
 	case constants.EphemeralPartitionLabel:
 		// Ephemeral Partition
-		typeID := "AF3DC60F-8384-7247-8E79-3D69D8477DE4"
+		typeID := "0FC63DAF-8483-4772-8E79-3D69D8477DE4"
 		opts = append(opts, partition.WithPartitionType(typeID), partition.WithPartitionName(t.Label), partition.WithMaximumSize(true))
 	default:
-		typeID := "AF3DC60F-8384-7247-8E79-3D69D8477DE4"
+		typeID := "0FC63DAF-8483-4772-8E79-3D69D8477DE4"
```

Comment:
Turns out that we have been doing the UUIDs wrong in the GPT package. 3 fields are actually middle endian (or mixed endiain). Which means the first three groups of the UUID should be encoded as Little Endian, while the rest should be encoded as Big Endian. When moving to grub we are required to have a specific partition type for BIOS.

```
Welcome to fdisk (util-linux 2.36).
Changes will remain in memory only, until you decide to write them.
Be careful before using the write command.


Command (m for help): p
Disk /dev/loop11: 4 GiB, 4294967296 bytes, 8388608 sectors
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 512 bytes / 512 bytes
Disklabel type: gpt
Disk identifier: 125A2162-E1D2-11EA-961D-025D91EBDA2C

Device          Start     End Sectors  Size Type
/dev/loop11p1      34 1048610 1048577  512M BIOS boot
/dev/loop11p2 1048611 8388574 7339964  3.5G Linux filesystem
```

Notice `BIOS boot` for the first partition's (boot partition) `Type`. Without the middle-endian UUID, this field is completely wrong. The `"AF3DC60F-8384-7247-8E79-3D69D8477DE4"` happened to be "correct" because it was set as such here in a hacky way that I can't remember, but the gist was a double negative. Best way I can explain it.